qiskit/transpiler/target.py

```diff
@@ -484,6 +484,29 @@ def operations_for_qargs(self, qargs):
             raise KeyError(f"{qargs} not in target.")
         return [self._gate_name_map[x] for x in self._qarg_gate_map[qargs]]
 
+    def instruction_supported(self, operation_name, qargs):
+        """Return whether the instruction (operation + qubits) is supported by the target
+
+        Args:
+            operation_name (str): The name of the operation for the instruction
+            qargs (tuple): The tuple of qubit indices for the instruction
+
+        Returns:
+            bool: Returns ``True`` if the instruction is supported and ``False`` if it isn't.
+
+        """
+        return_value = False
+        if operation_name in self._gate_map:
+            if qargs in self._gate_map[operation_name]:
+                return_value = True
+            elif self._gate_map[operation_name] is None:
+                if all(x < self.num_qubits for x in qargs):
+                    return_value = True
+            elif None in self._gate_map[operation_name]:
+                if all(x < self.num_qubits for x in qargs):
+                    return_value = True
```

Comment:
Does `Target` have any concept of how many arguments an operation should take that we should enforce in this case?

Reply:
It's not strictly enforced per operation, it's implied that the number of arguments is fixed and all entries in the target use the same number of arguments but nothing actively enforces it. I assume things would fail in some way if you do something like `target.add_instruction(XGate(), {(0,): None, (1,): None, (1, 2, 3): None})` since it's a baked in assumption that a operation in the target has a fixed number of arguments (like in some places the first qargs key for an operation is looked at to get the number of arguments quickly)

Reply:
That's fine if there's no separate internal tracking of the qargs count. I was just thinking about when the properties dict was None, or had a `None: None` element, and if there was going to be any other way to say that `instruction_supported("cx", (0, 1, 2, 3))` should be false.

Reply:
The only thing we can check is look at the gate instance with `target.operation_from_name(operation_name)` and compare the qarg count on that object (assuming it's an attribute of an instruction object, I'm not actually sure it is though) to the `len(qargs)`.

Reply:
Nah, if we don't have reliable access to the information, it's not important. As you say, it'll fail somewhere down the pipeline.

Reply:
I just checked and I think that if we did `len(qargs) == self.operation_from_name(operation_name).num_qubits` that would do the trick. I'll add this and a test for it too

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7778/commits/8bc78454d4f9c34efa16c150a9dc951b91dd077b